stan/math/prim/prob/wiener4_lcdf.hpp

```diff
@@ -0,0 +1,762 @@
+    const auto two_av = 2 * a * v;
+    const auto exp_two_va_one_minus_w = exp(two_va_one_minus_w);
+    const auto exp_two_avw = exp(two_avw);
+    const auto exp_two_av = exp(two_av);
+    if (((exp_two_va_one_minus_w >= nearly_one) || (exp_two_avw >= nearly_one))
+        || (exp_two_av >= nearly_one)) {
+      return ret_t(-w);
+    }
+    prob = LOG_TWO + two_va_one_minus_w - log1p(-exp_two_va_one_minus_w);
+    auto log_quotient = log1p(-exp_two_avw) - log1p(-exp_two_av);
+    if (log(w) > log_quotient) {
+      prob += log_diff_exp(log(w), log_quotient);
+      prob = exp(prob);
+    } else {
+      prob += log_diff_exp(log_quotient, log(w));
+      prob = -exp(prob);
+    }
+  } else {
+    const auto minus_two_va_one_minus_w = (-2.0 * v * a * (1.0 - w));
+    const auto minus_two_av = (-2 * a * v);
+    const auto exp_minus_two_va_one_minus_w = exp(minus_two_va_one_minus_w);
+    const auto exp_minus_two_av = exp(minus_two_av);
+    if ((exp_minus_two_va_one_minus_w >= nearly_one)
+        || (exp_minus_two_av >= nearly_one)) {
+      return ret_t(-w);
+    }
+    prob = LOG_TWO - log1p(-exp_minus_two_va_one_minus_w);
+    ret_t log_quotient;
+    if (minus_two_va_one_minus_w > minus_two_av) {
+      log_quotient = log_diff_exp(minus_two_va_one_minus_w, minus_two_av)
+                     - log1p(-exp_minus_two_av);
+    } else if (minus_two_va_one_minus_w < minus_two_av) {
+      log_quotient = log_diff_exp(minus_two_av, minus_two_va_one_minus_w)
+                     - log1p(-exp_minus_two_av);
+    } else {
+      log_quotient = NEGATIVE_INFTY - log1p(-exp_minus_two_av);
+    }
+    if (log(w) > log_quotient) {
+      prob += log_diff_exp(log(w), log_quotient);
+      prob = -exp(prob);
+    } else {
+      prob += log_diff_exp(log_quotient, log(w));
+      prob = exp(prob);
+    }
+  }
+  return (is_inf(prob)) ? NEGATIVE_INFTY : prob;
+}
+
+/**
+ * Calculate the wiener4 distribution
+ *
+ * @tparam NaturalScale Whether to return the distribution on natural or
+ * log-scale
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param vn The relative starting point
+ * @param wn The drift rate
+ * @param wildcard This parameter space is needed for a functor. Could be
+ * deleted when another solution is found
+ * @param err The log error tolerance
+ * @return distribution
+ */
+template <bool NaturalScale = false, typename T_y, typename T_a, typename T_w,
+          typename T_v, typename T_wildcard, typename T_err>
+inline auto wiener4_distribution(const T_y& y, const T_a& a, const T_v& vn,
+                                 const T_w& wn, T_wildcard&& wildcard = 0.0,
+                                 T_err&& err = log(1e-12)) noexcept {
+  using ret_t = return_type_t<T_y, T_a, T_w, T_v>;
+  ret_t log_distribution = NEGATIVE_INFTY;
+  const auto v = -vn;
+  const auto w = 1 - wn;
+
+  if (is_inf(y)) {
+    return ret_t(log_probability_distribution(a, v, w));
+  }
```

Comment:
If y is infinity here why is it removed from the calculation?

Reply:
I removed these lines as there is a check at the beginning of the main function that `y` has to be positive-finite.